cirq/circuits/circuit.py

```diff
@@ -2213,18 +2213,11 @@ def decompose(self, add_empty_moments=True) -> Iterable['cirq.Circuit']:
         for op in self.all_operations():
             if len(op.qubits) > 1:
                 uf.union(*op.qubits)
-        qubit_sets = sorted(uf.to_sets(), key=lambda x: min(x))
+        qubit_sets = sorted(uf.to_sets(), key=min)
         if len(qubit_sets) == 1:
             return (self,)
-        else:
-
-            new_moments: List[List['cirq.Moment']] = [[] for _ in qubit_sets]
-            for m in self.moments:
-                for mi, qubits in enumerate(qubit_sets):
-                    new_moment = m[qubits]
-                    if add_empty_moments or len(new_moment) > 0:
-                        new_moments[mi].append(m[qubits])
-            return (Circuit(m) for m in new_moments)
+        new_moments = [[m[qubits] for m in self.moments] for qubits in qubit_sets]
+        return (Circuit(m) for m in new_moments)
```

Comment:
nit: it seems a bit odd to me to construct the moments eagerly but then only construct the circuits lazily. I'd suggest doing the whole thing in a single comprehension, and I lean slightly toward eagerness since `Circuit` is mutable so could in theory change between the time you call `decompose` and actually run the generator:
```python
return [Circuit(m[qubits] for m in self.moments) for qubits in qubit_sets]
```

Reply:
Good point, done.

Reply:
did you not push this change? I'm not seeing it

Reply:
I did - that's how line 2230 looks for me when I look at changed files in github PR:

```
return (Circuit(m[qubits] for m in self.moments) for qubits in qubit_sets)
```

Reply:
If the problem persists, can you let me know what is the latest commit visible to you?